### PR TITLE
Entering EMIS in wrong twice kicks user to endstate

### PR DESCRIPTION
**Pull request by imsickofmaps**

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@smn ready for review. Before you comment, I did try and make it a bit DRYer... but because there's a few different flows that involve the user entering/changing EMIS with various user states it become too complex to easily validate in tests locally.

File: js_sandbox/lib/go-rts-zambia.js [hunk 1]
```diff
@@ -615,37 +762,39 @@ function GoRtsZambia() {
...
+    //             "reg_school_teachers",
+    //             "How many classrooms do you have in your school?",
+    //             function(content) {
+    //                 // check that the value provided is actually decimal-ish.
+    //                 return !Number.isNaN(Number(content));
+    //             },
+    //             'Please provide a number value for how many classrooms you have in your school.'
+    //         )
+    //     }
+    // });
```
Comment (smn) [inline js_sandbox/lib/go-rts-zambia.js]: Please remove? Commented code is dead code, if we ever need it again we can pull it out of Git's history.

File: js_sandbox/lib/go-rts-zambia.js [hunk 2]
```diff
@@ -615,37 +762,39 @@ function GoRtsZambia() {
...
+
+    self.add_state(new FreeText(
+        "reg_school_classrooms",
+        "reg_school_teachers",
+        "How many classrooms do you have in your school?",
+        function(content) {
+            // check that the value provided is actually decimal-ish.
+            return !Number.isNaN(Number(content));
```
Comment (smn) [inline js_sandbox/lib/go-rts-zambia.js]: This fails on blank input.

**Comment by smn** [comment]
:+1: in general. This is a good example for how a violation of the http://en.wikipedia.org/wiki/Zero_one_infinity_rule makes things unnecessarily complex.

**Pull request merged**